src/types/ScholarshipAmount.ts

```diff
@@ -51,7 +51,7 @@ export default class ScholarshipAmount {
           throw new Error(`Invalid range ${min}-${max}`);
         }
         this.min = min ?? 0;
-        this.max = max ?? RANGE_MAX;
+        this.max = max || RANGE_MAX;
```

Comment:
Ah I see what you meant now. `max ?? RANGE_MAX` doesn't set `RANGE_MAX` if `max` is set to anything (including 0) but `max || RANGE_MAX` sets the `RANGE_MAX` on 0.

Reply:
yes, that's why I had 3 cases checking if min,max or both were set lol